### PR TITLE
✨ Add book description and locale support

### PR DESCRIPTION
**Pull request by williamchong**
Base on https://github.com/likecoin/likecoin-api-public/pull/472

### CODE REVIEW

File: src/util/ses.ts [hunk 1]
```diff
@@ -328,3 +328,41 @@ export function sendNFTBookPendingClaimEmail({
...
+        Value: 'sendNFTBookPendingClaimEmail',
+      },
+    ],
+    Destination: {
+      ToAddresses: emails,
+    },
+    Message: {
+      Subject: {
+        Charset: 'UTF-8',
+        Data: `You have sold an nft for $${amount}`,
```
Comment (nwingt) [inline src/util/ses.ts]:
```suggestion
        Data: `You have sold an NFT for $${amount}`,
```

File: src/util/api/likernft/book/index.ts [hunk 2]
```diff
@@ -110,6 +154,14 @@ export async function processNFTBookPurchase(
...
       });
     }
+    if (notificationEmails.length) {
+      await sendNFTBookSalesEmail({
+        buyerEmail: email,
+        emails: notificationEmails,
+        classId,
+        amount: (amountTotal || 0) / 100,
+      });
+    }
```
Comment (nwingt) [inline src/util/api/likernft/book/index.ts]: I think we should also alert the seller when a buyer claimed the NFT, so the seller can take action to sign and send the NFT

File: src/routes/likernft/book/store.ts [hunk 3]
```diff
@@ -160,19 +161,36 @@ router.post('/:classId/new', jwtAuth('write:nftbook'), async (req, res, next) =>
...
       throw new ValidationError(`INVALID_PRICE_in_${invalidPriceIndex}`);
     }
+    const invalidNameIndex = prices.findIndex((p) => {
+      const {
+        name = {},
+        description = {},
+      } = p;
+      return !(
+        typeof name[NFT_BOOK_TEXT_DEFAULT_LOCALE] === 'string'
+        && (description[NFT_BOOK_TEXT_DEFAULT_LOCALE] && typeof description[NFT_BOOK_TEXT_DEFAULT_LOCALE] === 'string'));
```
Comment (nwingt) [inline src/routes/likernft/book/store.ts]: We just check the default language?

File: src/util/api/likernft/book/index.ts (hunk 2 above)
Comment (williamchong) [reply src/util/api/likernft/book/index.ts]: Not sure if we want to alert both on sale and claim, but makes sense

File: src/routes/likernft/book/store.ts (hunk 3 above)
Comment (williamchong) [reply src/routes/likernft/book/store.ts]: you mean checking string for all locale that exists as a key in `description`?

File: src/util/ses.ts (hunk 1 above)
Comment (williamchong) [reply src/util/ses.ts]: handle in https://github.com/likecoin/likecoin-api-public/pull/476

File: src/util/api/likernft/book/index.ts (hunk 2 above)
Comment (williamchong) [reply src/util/api/likernft/book/index.ts]: handle in https://github.com/likecoin/likecoin-api-public/pull/476

File: src/routes/likernft/book/store.ts (hunk 3 above)
Comment (nwingt) [reply src/routes/likernft/book/store.ts]: Yes, or do we want to make sure the default exists?

Comment (williamchong) [reply src/routes/likernft/book/store.ts]: we do want to make sure at least the default exists, but we probably want to make sure any other key that exists is also a string

**Review by nwingt — APPROVED** [review]

**Pull request merged**